app/models/vacation.rb

```diff
@@ -1,10 +1,21 @@
 class Vacation < ApplicationRecord
 
   belongs_to :employee
+  belongs_to :position
   
   validates :start_date, presence: true
   validates :end_date, presence: true
+  validates :position_id, presence: true
   validate :end_date_after_start_date
+  validate :validate_department_vacation_limit
+
+  def vacation_period
```

Comment:
This method doesn't really have the right place. The model side should have everything related to CRUD operations. I would move it to the decorator.